WordPress/Classes/Models/ReaderPost+Helper.swift

```diff
@@ -0,0 +1,22 @@
+import Foundation
+
+extension ReaderPost {
+
+    /// Find cached comment with given ID.
+    ///
+    /// - Parameter id: The comment id
+    /// - Returns: The `Comment` object associated with the given id.
+    @objc
+    func comment(withID id: NSNumber) -> Comment? {
+        comment(withID: id.int32Value)
+    }
+
+    /// Find cached comment with given ID.
+    ///
+    /// - Parameter id: The comment id
+    /// - Returns: The `Comment` object associated with the given id.
+    func comment(withID id: Int32) -> Comment? {
+        return (comments as? Set<Comment>)?.first { $0.commentID == id }
+    }
+
+}
```

Comment:
This logic is clearly duplicated between `ReaderPost` and `Blog`. I wonder if there's a way to extract it in a protocol and protocol extension?

```swift
protocol /* A name representing "having comments" */ {
  var comments: NSSet<Comment> { get }  // would that work as a type definition?
}

extension /* A name representing "having comments" /* {
  ...
  func comment(withID id: Int32) -> Comment? { ... }
}
```

---

Update: After looking at the rest of the PR, I doubt this would work in the Objective-C part of the code. If you 1) agree with the DRY suggestion and 2) can't find a way to make it work either (or don't have time to), would it be worth leaving a note in both extensions to make the other known and mention that we might be able to centralize them once we get rid of the Objective-C code?

Reply:
IMO, the implementation is so simple that I feel like it's not worth the effort of abstraction? If we were to make an abstraction, I probably would go for adding an extension method, rather than creating a new protocol. Something like:
```swift
// pseudo code
extension Set /* or Sequence where Value: Equatable */ {
    func first(keyPath: KeyPath<Self, Value>, equals value: Value) -> Element {
         first(where: { $0[keyPath: keyPath] == value })
    }
}

blog.comments.first(\.commentID, equals: id)
post.comments.first(\.commentID, equals: id)
```

Reply:
> IMO, the implementation is so simple that I feel like it's not worth the effort of abstraction?

Fair enough. I also subscribe to the [three strikes then automate (or, in this case, abstract) rule](https://wiki.c2.com/?ThreeStrikesAndYouAutomate)

> - The first time you do something, you just do it manually.
> - The second time you do something similar, you wince at the repetition, but you do it anyway.
> - The third time you do something similar, you automate.

---

I like the `Set` `extension` idea, because it makes the resulting code more general purpose. I can't think of another use case for that functionality, but it would make for neat Swift 😄